tests/ci/build_report_check.py

```diff
@@ -231,7 +231,8 @@ def main():
     total_groups = len(build_results)
     logging.info("Totally got %s artifact groups", total_groups)
     if total_groups == 0:
-        fail_simple_check(gh, pr_info, f"{build_check_name} failed")
+        if needs_data and any(i["result"] != "skipped" for i in needs_data.values()):
```

Comment:
Comment here will be appreciated by future us

```suggestion
        # A report might be empty in case of `do not test` label, for example.
        # We should still be able to merge such PRs
        if needs_data and any(i["result"] != "skipped" for i in needs_data.values()):
```